clients/drcachesim/simulator/cache_fifo.h

```diff
@@ -52,6 +52,8 @@ class cache_fifo_t : public cache_t {
     access_update(int line_idx, int way) override;
     int
     replace_which_way(int line_idx) override;
+    int
+    get_next_way_to_replace(int block_idx);
```

Comment:
There's no use of this function.  Generally we'd like a test or use of a new function in the same PR: it seems unusual to just add a function without anything calling it.  Maybe it is best to add this code to the test PR which presumably uses this function.

Or, if replace_which_way can have side effects, maybe this const query should be in the base class, and the LRU test should use it instead of replace_which_way, in this PR.

Reply:
It's being used by this draft PR: https://github.com/DynamoRIO/dynamorio/pull/5454 but agree it makes sense to put it in that PR in the base class. Will do that.

Reply:
Actually it can't be in the base class as each cache replacement algorithm's replace policy is different.

Reply:
> Or, if replace_which_way can have side effects, maybe this const query should be in the base class, and the LRU test should use it instead of replace_which_way, in this PR.

`replace_which_way()` implementation in LRU cache doesn't have a side effect as of the current LRU cache test cases.

Reply:
> Actually it can't be in the base class as each cache replacement algorithm's replace policy is different.

This *implementation* wouldn't be in the base class, but a virtual method would be, and each subclass would override it with their particular implementation.  All tests would then call this new const method, rather than the one that can have side effects, as the cleaner and safer way to query.

Reply:
Understood. Will make that change in PR https://github.com/DynamoRIO/dynamorio/pull/5454.